apps/core/components/userContext/UserContext.js

```diff
@@ -23,9 +41,58 @@ export default class UserContextProvider extends React.Component<Props, State> {
 
     this.state = {
       ...defaultState,
+      actions: {
+        signIn: this.signIn,
+        signOut: this.signOut,
+      },
     };
   }
 
+  componentDidMount() {
+    if (this.state.isFirebaseAvailable) {
+      this.fireBaseListener = firebase.auth().onAuthStateChanged(authUser => {
+        this.setState({
+          isUserSignedIn: authUser != null,
+          ...this.mapUserData(authUser),
+        });
+      });
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.fireBaseListener) {
+      /**
+       * NOTE: Auth listener is removed by calling the unsubscribe
+       * function returned by `onAuthStateChanged`
+       */
+      this.fireBaseListener();
+    }
+  }
+
+  fireBaseListener: () => void;
+
+  mapUserData = (authUser: Object) => {
+    return {
+      userEmail: authUser?.email,
+      userDisplayName: authUser?.displayName,
+    };
+  };
+
+  signIn = () => {
+    signInWithGoogle().catch(() => {
+      // @TODO Handle sign-in error - show notification or message
+    });
+  };
+
+  signOut = () => {
+    firebase
+      .auth()
+      .signOut()
+      .catch(() => {
+        // @TODO Handle sign-out error - show notification or message
+      });
```

Comment:
signIn and signOut looks very much like promises. Are they? And if so, can we use async/await instead?

Reply:
Sure, updated. I'll probably even move whole try...catch part directly to SignIn/SignOut component to handle UI changes during login. But I'm saving this for next PR to keep this one smaller.